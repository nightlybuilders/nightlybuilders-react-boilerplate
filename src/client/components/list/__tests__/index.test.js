import React from 'react'
import { shallow } from 'enzyme'
import map from 'lodash.map'

import { List, ListItem } from '../'

describe('Component/List', () => {
  const listItems = [{ id: 1, title: 'some item' }, { id: 2, title: 'some other item' }]

  test('should render without throwing an error', () => {
    const wrapper = shallow(
      <List>{map(listItems, item => <ListItem key={item.id}>{item.title}</ListItem>)}</List>,
    )
    expect(wrapper).toMatchSnapshot()
  })

  test('should attach an onClick and onKeyPress event to ListItems', () => {
    const onClick = jest.fn()
    const onKeyPress = jest.fn()

    const wrapper = shallow(
      <List>
        {map(listItems, item => (
          <ListItem key={item.id} onClick={onClick} onKeyPress={onKeyPress}>
            {item.title}
          </ListItem>
        ))}
      </List>,
    )
    const firstItem = wrapper.find(ListItem).first()

    firstItem.simulate('click')
    firstItem.simulate('keypress')
    expect(onClick).toHaveBeenCalledTimes(1)
    expect(onKeyPress).toHaveBeenCalledTimes(1)
  })
})
