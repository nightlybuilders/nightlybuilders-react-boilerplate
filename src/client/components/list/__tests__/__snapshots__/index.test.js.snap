// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component/List should render without throwing an error 1`] = `
<ul
  className="list"
>
  <ListItem
    className=""
    key="1"
    onClick={[Function]}
    onKeyPress={[Function]}
    style={Object {}}
  >
    some item
  </ListItem>
  <ListItem
    className=""
    key="2"
    onClick={[Function]}
    onKeyPress={[Function]}
    style={Object {}}
  >
    some other item
  </ListItem>
</ul>
`;
